lms/djangoapps/edraak_forus/tests/tests.py

```diff
@@ -0,0 +1,388 @@
+            User.objects.get(email=self.user_email)
+
+    def test_closed_course(self):
+        # pylint: disable=invalid-name
+        with self.assertRaisesRegexp(ValidationError, 'Enrollment.*closed.*go.*ForUs') as cm:
+            self._validate_params(course_id=unicode(self.closed_course.id))
+
+        self._assertErrorCount(cm.exception, 1)
+
+    def test_current_course(self):
+        try:
+            self._validate_params(course_id=unicode(self.current_course.id))
+        except ValidationError as exc:
+            self.fail('The course is open and everything is fine, yet there is an error: `{}`'.format(exc))
+
+    def test_upcoming_course(self):
+        try:
+            self._validate_params(course_id=unicode(self.upcoming_course.id))
+        except ValidationError as exc:
+            self.fail('The course is upcoming and everything is fine, yet there is an error: `{}`'.format(exc))
+
+    def test_draft_course(self):
+        # pylint: disable=invalid-name
+        with self.assertRaisesRegexp(ValidationError, '.*not.*opened.*go.*ForUs') as cm:
+            self._validate_params(course_id=unicode(self.draft_course.id))
+
+        self._assertErrorCount(cm.exception, 1)
+
+    @ddt.data('', ' ', 'XZ')
+    def test_invalid_country(self, bad_value):
+        self._assertValidateData(
+            field_id='country',
+            bad_value=bad_value,
+            exception_regexp='.*Invalid.*country.*',
+        )
+
+    @ddt.data('', ' ')
+    def test_invalid_name(self, bad_value):
+        self._assertValidateData(
+            field_id='name',
+            bad_value=bad_value,
+            exception_regexp='.*Invalid.*name.*',
+        )
+
+    @ddt.data('', ' ', 'hello')
+    def test_invalid_level_of_education(self, bad_value):
+        self._assertValidateData(
+            field_id='level_of_education',
+            bad_value=bad_value,
+            exception_regexp='.*Invalid.*level of education.*',
+        )
+
+    @ddt.data('', ' ', 'o', 'hello')
+    def test_invalid_gender(self, bad_value):
+        self._assertValidateData(
+            field_id='gender',
+            bad_value=bad_value,
+            exception_regexp='.*Invalid.*gender.*',
+        )
+
+    def test_invalid_hmac(self):
+        self._assertValidateData(
+            field_id='forus_hmac',
+            bad_value='wrong_dummy_hmac',
+            exception_regexp='The security check has failed on the provided parameters'
+        )
+
+    def test_missing_hmac(self):
+        params = build_forus_params(email=self.user_email, course_id=unicode(self.upcoming_course.id))
+        params.pop('forus_hmac', None)
+        # pylint: disable=invalid-name
+        with self.assertRaisesRegexp(ValidationError, 'The security check has failed on the provided parameters') as cm:
+            ValidateForusParams(params).validate()
+
+        self.assertIn('forus_hmac', cm.exception.message_dict)
+
+    def test_calculate_hmac(self):
```

Comment:
Not enough my friend. We need to have some hard data.

The code below merely duplicates the function logic, which is not good for testing. Fixtures is a better solution here.

Try to generate the fixtures using an external tool like bash. However the fixtures should be hardcoded, and the bash commands (for example) should be documented.